d3a_api_client/redis_client_base.py

```diff
@@ -145,14 +125,42 @@ def _on_unregister(self, msg):
             raise RedisAPIException(f'Failed to unregister from market {self.area_id}.'
                                     f'Deactivating connection.')
 
-    def _check_buffer_message_matching_command_and_id(self, message):
-        if key_in_dict_and_not_none(message, "transaction_id"):
-            transaction_id = message["transaction_id"]
-            if not any(command in ["register", "unregister"] and "transaction_id" in data and
-                       data["transaction_id"] == transaction_id
-                       for command, data in self._blocking_command_responses.items()):
-                raise RedisAPIException("There is no matching command response in _blocking_command_responses.")
-        else:
-            raise RedisAPIException(
-                "The answer message does not contain a valid 'transaction_id' member.")
+    def _on_event_or_response(self, msg):
+        message = json.loads(msg["data"])
+        function = lambda: self.on_event_or_response(message)
+        self.executor.submit(execute_function_util, function=function,
+                             function_name="on_event_or_response")
+
+    def select_aggregator(self, aggregator_uuid, is_blocking=True):
+
+        if not self.area_uuid:
+            assert False
```

Comment:
Nit: it would be better to explicitly raise a custom exception rather than using this strategy to raise an `AssertionError`. Since this is mostly refactoring of existing code, it's not necessary to deal with this in this PR anyway.

Reply:
Indeed, will add it.